From 1835e0c82fca788aef823d91bda4fedef26cd712 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 1 Aug 2018 10:10:16 +0100
Subject: [PATCH] Resolves: rhbz#1610692 rectangles ctor takes topleft,
 bottomright points

so pass arguments in the topleft, bottomright order to avoid eventual...

Gtk-CRITICAL **: 09:43:33.896: gtk_widget_queue_draw_area: assertion 'height >= 0' failed

Change-Id: I2396f49470274331c49275a5e6d707f59277be4a
Reviewed-on: https://gerrit.libreoffice.org/58392
Tested-by: Jenkins
Reviewed-by: Michael Stahl <Michael.Stahl@cib.de>
---
 vcl/source/control/fixed.cxx | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/vcl/source/control/fixed.cxx b/vcl/source/control/fixed.cxx
index dac10a896278..eeba87afab05 100644
--- a/vcl/source/control/fixed.cxx
+++ b/vcl/source/control/fixed.cxx
@@ -531,11 +531,11 @@ void FixedLine::ImplDraw(vcl::RenderContext& rRenderContext)
         rRenderContext.DrawText(aTextPt, aText, 0, aText.getLength());
         rRenderContext.Pop();
         if (aOutSize.Height() - aStartPt.Y() > FIXEDLINE_TEXT_BORDER)
-            aDecoView.DrawSeparator(Point(aStartPt.X(), aOutSize.Height() - 1),
-                                    Point(aStartPt.X(), aStartPt.Y() + FIXEDLINE_TEXT_BORDER));
+            aDecoView.DrawSeparator(Point(aStartPt.X(), aStartPt.Y() + FIXEDLINE_TEXT_BORDER),
+                                    Point(aStartPt.X(), aOutSize.Height() - 1));
         if (aStartPt.Y() - nWidth - FIXEDLINE_TEXT_BORDER > 0)
-            aDecoView.DrawSeparator(Point(aStartPt.X(), aStartPt.Y() - nWidth - FIXEDLINE_TEXT_BORDER),
-                                    Point(aStartPt.X(), 0));
+            aDecoView.DrawSeparator(Point(aStartPt.X(), 0),
+                                    Point(aStartPt.X(), aStartPt.Y() - nWidth - FIXEDLINE_TEXT_BORDER));
     }
     else
     {
-- 
2.17.1

